### ISSUE TITLE
Review triggering is soft while capture is hard-enforced, so the backlog grows silently (follow-up to #10)

### ISSUE THREAD

**Issue by michaelkariuki** [rebelytics/one-skill-to-rule-them-all#34]
Follow-up to #10, which reported the same symptom ("the first review never fired... by then I had about 30 notes sitting there unresolved") and was addressed by having setup write `never` into `last-review-date.txt` so the value is meaningful.

My setup had that fix in place and behaved correctly at the file level — and the review still never ran. I think the file value was the visible half of the problem, and the trigger's *enforcement tier* is the other half.

## What I found

Two weeks after install, across two unrelated projects:

- 15 observations logged, all well-formed, every one carrying `**Status:** OPEN`
- `last-review-date.txt` = `never` in both workspaces (correct — no review had run)
- `archive/` empty in both
- `cross-cutting-principles.md` present but still the bare template in both
- 4 new-skill candidates named across the observations; **none created**

So the capture half worked exactly as designed for two weeks, and the consume half ran zero times. The observations themselves were good — that is what makes the failure mode worth reporting rather than just my own inattention.

## Why I think the two halves diverged

The asymmetry looks structural rather than accidental:

- **Capture** is enforced by hard checkpoints hooked onto tool calls the agent is already making — the "every 3rd TodoWrite completion" rule and the deliverable-event flush. `SKILL.md` is explicit that this is deliberate: *"Hard enforcement that hooks onto tool calls you are already making is the only reliable mechanism; soft prompts that rely on memory don't survive cognitive load."*
- **The review** is triggered by Session Start step 3 — read a file, compare a date, offer a review in one line. That is precisely the soft, memory-dependent prompt the capture rules were designed not to rely on. It sits at the start of a session, competing with the user's actual request, and it is skipped the same way activation is skipped in #20 and #22.

The reasoning that justifies the capture checkpoints applies unchanged to the review, but it was not carried across.

What makes this worse than a missed activation is the failure signature: it is silent and self-concealing. Capture keeps producing, so the log looks healthy and *growing*. Nothing errors. The only artefact that records the failure is a file reading `never` that, by construction, nobody is reading — that is the very step that did not fire. From inside any session, an unreviewed 15-entry backlog is indistinguishable from a well-maintained one.

## Suggested change

Treat "is the review trigger structurally enforced?" as an install-completeness check, the same way the skill already treats capture checkpoints — and ship a setup step that installs a host-level session-start trigger rather than relying on the agent to read the timestamp.

The trigger should compute the state from the files and inject it, rather than instructing the agent to go and look. In Claude Code that is a `SessionStart` hook returning `hookSpecificOutput.additionalContext`; the equivalent exists in most harnesses. The essential logic is small:

```bash
headers=$(grep -c '^### Observation ' "$LOG")
resolved=$(grep -cE '^\*\*Status:\*\* (ACTIONED|DECLINED)' "$LOG")
open_count=$(( headers - resolved ))
# ...compare last-review-date.txt against today, then emit the count and the
# days-since-review as context, with the review offer already composed
```

Deriving the OPEN count as *headers minus resolved* also matches the reconciliation guard in `references/weekly-review.md` — statusless entries count as OPEN rather than vanishing from a `grep '**Status:** OPEN'`.

Two details that mattered when I built this:

1. **`grep -c` exits 1 on zero matches but still prints `0`.** Writing `$(grep -c ... || echo 0)` concatenates two values and yields `0\n0`, which then breaks the arithmetic. Capture the output and ignore the exit code.
2. **The reminder branch needs to be proven to fire.** I only trusted mine after running it against fixtures at `never`, 30 days stale, and 2 days stale, and confirming the third stays silent — a nag that never fires and a nag that is correctly silent look identical from a passing run.

This does not make invocation certain — as #22 notes, a hook can only inject a prompt and the model still decides. But it moves the review from "the agent remembers to check a file" to "the number is already in context," which is the same upgrade the capture checkpoints made.

## Generalised principle

In any loop that captures material for later processing, the enforcement asymmetry between capture and consume decides the outcome — and a soft consume step does not degrade gracefully. It degrades to zero while capture keeps producing, so the backlog grows and the system looks healthy from the inside. Whatever mechanism makes capture reliable has to be applied to consumption; a reminder the agent has to remember to read is not a trigger.

---

*Filed after a consolidation pass that merged the backlog and actioned all 15 entries. Happy to share the hook script if it's useful as a starting point.*

**Comment by rebelytics** [comment]
Shipped in v3.0.0, released today: a sample SessionStart hook in references/environments.md computes the open count and review staleness and injects them, with both of your gotchas covered. You're credited as co-author on the commit. Thanks again for your contribution.
